packages/zendesk-support-adapter/src/filters/handle_template_expressions.ts

```diff
@@ -125,29 +199,42 @@ const filterCreator: FilterCreator = () => {
   return ({
     onFetch: async (elements: Element[]): Promise<void> => log.time(async () =>
       replaceFormulasWithTemplates(elements.filter(isInstanceElement)), 'Create template creation filter'),
-    preDeploy: (changes: Change<InstanceElement>[]): Promise<void> => log.time(async () =>
+    preDeploy: (changes: Change<InstanceElement>[]): Promise<void> => log.time(async () => {
       (await getContainers(await awu(changes).map(getChangeData).toArray())).forEach(
         async container => {
           const { fieldName } = container.template
           const handleTemplateValue = (template: TemplateExpression): string => {
             const templateUsingIdField = new TemplateExpression({
-              parts: template.parts.map(part => (isReferenceExpression(part)
-                ? [saltoTypeToZendeskReferenceType[part.elemID.typeName],
-                  '_',
-                  new ReferenceExpression(part.elemID.createNestedID('id'), part.value.value.id)]
-                : part)).flat(),
+              parts: template.parts.map(part => {
+                if (isReferenceExpression(part)) {
+                  if (saltoTypeToZendeskReferenceType[part.elemID.typeName]) {
+                    return [saltoTypeToZendeskReferenceType[part.elemID.typeName],
+                      '_',
+                      new ReferenceExpression(part.elemID.createNestedID('id'), part.value.value.id)]
```

Comment:
styling nit - this is pretty hard to read, maybe
```suggestion
                    return [
                      saltoTypeToZendeskReferenceType[part.elemID.typeName],
                      '_',
                      new ReferenceExpression(part.elemID.createNestedID('id'), part.value.value.id),
                    ]
```